packages/db/index.js

```diff
@@ -484,8 +499,13 @@ export class DBClient {
       query = query.gte('inserted_at', opts.after)
     }
 
-    /** @type {{ data: Array<import('./db-client-types').UploadItem>, error: Error }} */
-    const { data: uploads, error } = await query
+    // Apply pagination or limiting.
+    query = opts.offset
+      ? query.range(offset, limit - 1)
```

Comment:
Looks like this is what I want, but curious as to if anyone knows a better way of doing this? 
Am unsure of the performance impact of using a range()...

Reply:
AFAIK the `.range()` method translates to the SQL `LIMIT a OFFSET b`. So I think it should be fine. The only other way I can think of doing it would be if we used the column that we're sorting by (e.g. `inserted_at`) and took the highest value from the previous page and then did `query.gt('inserted_at', prev_highest)`, which would presumably be very very sliiiightly faster because it would save the DB from having to count and skip the first `offset` rows that it finds. But given that we're only talking about the uploads for a single user here, I can't imagine that we're likely to get to the number of rows where this difference actually matters.

What probably _is_ worth doing is checking that we've got indexes for the columns that we're filtering and sorting on.